airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py

```diff
@@ -0,0 +1,265 @@
+        self._start_date = start_date
+        super().__init__(**kwargs)
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        resp_json = response.json()
+        links = resp_json.get("links", {})
+        if links.get("next"):
+            next_query_string = urllib.parse.urlsplit(links["next"]).query
+            params = dict(urllib.parse.parse_qsl(next_query_string))
+            return params
+
+    def request_headers(self, **kwargs) -> Mapping[str, Any]:
+        return {"Content-Type": "application/json"}
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        if response_json.get("error"):
+            raise Exception(repr(response_json.get("error")))  # TODO: apply ShouldRetry
+        result = response.json.get(self.data_field, [])
+        yield from result
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = {}
+        if next_page_token:
+            params.update(next_page_token)
+        return params
+
+    def read_records(  # TODO remove this (caching) in final commit
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        with vcr.use_cassette("pretty_name.yaml", record_mode="new_episodes"):
+            yield from super().read_records(
+                sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state
+            )
+
+
+class IncrementalSurveymonkeyStream(SurveymonkeyStream, ABC):
+
+    state_checkpoint_interval = 1000
+
+    @property
+    @abstractmethod
+    def cursor_field(self) -> str:
+        """
+        Defining a cursor field indicates that a stream is incremental, so any incremental stream must extend this class
+        and define a cursor field.
+        """
+
+    def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["sort_order"] = "ASC"
+        params["sort_by"] = "date_modified"
+        params["per_page"] = 1000  # maybe as user input or bigger value
+        since_value = pendulum.parse(stream_state.get(self.cursor_field)) if stream_state.get(self.cursor_field) else self._start_date
+
+        since_value = max(since_value, self._start_date)
+        params["start_modified_at"] = since_value.strftime("%Y-%m-%dT%H:%M:%S")
+        return params
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
```

Comment:
Also since you have only 2 incremental streams why do you have this `get_updated_state` function in `IncrementalSurveymonkeyStream` class and not in `Surveys` class? Why did you decide that this function will be as base in `IncrementalSurveymonkeyStream`? Why didn't you place here `get_updated_state` from `SurveyResponses`?

Reply:
it has its own state too, its state and parent state are different subjects